### ISSUE TITLE
ESLint API ignores configured globals in no-undef and no-global-assign

### ISSUE THREAD

**Issue by fireairforce** [utooland/utoo-lint#1212]
Configuring a custom readonly global through the ESLint-compatible API produces both a false positive in `no-undef` and a false negative in `no-global-assign`.

Reproduced with `@utoo/lint@0.4.0`, Node.js 22.23.1, macOS arm64. The same behavior is present in a local checkout of commit `72dacd385fea41816e1dcf6cb7f479151faadfa1`.

### Reproduction

In an empty directory:

```sh
npm init -y
npm install @utoo/lint@0.4.0 eslint@8.57.1
```

Save this as `repro.cjs` and run `node repro.cjs`:

```js
const original = require('eslint');
const replacement = require('@utoo/lint');

(async () => {
  for (const [name, { ESLint }] of [
    ['eslint', original],
    ['@utoo/lint', replacement]
  ]) {
    for (const [rule, source] of [
      ['no-undef', 'allowed(); missing();'],
      ['no-global-assign', 'allowed = 1;']
    ]) {
      const engine = new ESLint({
        useEslintrc: false,
        overrideConfig: {
          parserOptions: { ecmaVersion: 2022 },
          globals: { allowed: 'readonly' },
          rules: { [rule]: 'error' }
        }
      });
      const [result] = await engine.lintText(source, {
        filePath: 'input.js'
      });
      console.log(name, rule, result.messages.map(
        ({ ruleId, message }) => ({ ruleId, message })
      ));
    }
  }
})();
```

### Expected

As reported by ESLint 8:

- `no-undef`: only `missing` is undefined.
- `no-global-assign`: assigning to `allowed` reports a readonly-global error.

### Actual

With `@utoo/lint`:

- `no-undef`: both `allowed` and `missing` are reported as undefined.
- `no-global-assign`: no diagnostic is reported for `allowed = 1`.

This makes explicit project globals unreliable when using the compatibility API. Please preserve both the existence and readonly/writable state of configured globals when running native rules.
